### PR TITLE
Raster values to mesh scalars

### PR DESCRIPTION
**Pull request by rinkk**
this got a little out of hand. i basically rewrote the VtkMeshConverter parts for converting images to meshes. with the tiny addition that you can now choose if pixel values should be interpreted as mat ids or double arrays.
there will be at least two follow-up PRs on this but i guess this one is already long enough.

### CODE REVIEW

**Comment by endJunction** [comment]
Didn't check the logic.
There are fewer explicit memory allocations and frees: :+1: 

:fast_forward: after whitespace cleanup.

File: Applications/DataExplorer/VtkVis/VtkVisPipelineView.h [hunk 1]
```diff
@@ -56,6 +56,13 @@ protected slots:
 	/// Creates a menu on right-clicking on an item.
 	void contextMenuEvent(QContextMenuEvent* event);
 
+	/// Calls the conversion method for creating an OGS Mesh from a vtkImageData object.
+	void constructMeshFromImage(
+		std::string msh_name,
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkVisPipelineView.h]: Why not use a `const &` for the `msh_name`?

File: MeshLib/MeshGenerators/VtkMeshConverter.h [hunk 2]
```diff
@@ -72,19 +73,66 @@ class VtkMeshConverter
...
+		std::vector<double> const& pix_val,
+		std::vector<bool> const& pix_vis,
+		const std::size_t &imgHeight,
+		const std::size_t &imgWidth,
+		MeshElemType elem_type)
+	{
+		for (std::size_t i = 0; i < imgWidth; i++)
+			for (std::size_t j = 0; j < imgHeight; j++)
+			{
+				std::size_t const idx (i*imgHeight+j);
```
Comment (TomFischer) [inline MeshLib/MeshGenerators/VtkMeshConverter.h]: I do not see where `idx` is used. You could use it in line 123.

**Comment by endJunction** [comment]
Still some trailing whitespaces.
```sh
> git diff --check `git merge-base ufz/master rk/RasterValAsScalar` rk/RasterValAsScalar
MeshLib/MeshGenerators/VtkMeshConverter.cpp:95: trailing whitespace.
+                       double const value = (nTuple < 3) ? 
MeshLib/MeshGenerators/VtkMeshConverter.cpp:105: trailing whitespace.
+       
MeshLib/MeshGenerators/VtkMeshConverter.cpp:172: trailing whitespace.
+               fillPropertyVector<int>(*prop_vec, pix_val, pix_vis, imgHeight, imgWidth, elem_type); 
MeshLib/MeshGenerators/VtkMeshConverter.cpp:173: trailing whitespace.
+       }       
MeshLib/MeshGenerators/VtkMeshConverter.cpp:178: trailing whitespace.
+               fillPropertyVector<double>(*prop_vec, pix_val, pix_vis, imgHeight, imgWidth, elem_type); 
```

**Pull request merged**